app/api/crud.py

```diff
@@ -207,27 +208,37 @@ async def get(
                             "session_type",
                             "pipeline_name",
                         ],
-                        dropna=True,
-                    )
-                    .agg(
+                        # Keep NaNs to ensure that when there are no pipeline_name values in the query result,
+                        # we don't end up with an empty dataframe for pipeline_grouped_data
+                        dropna=False,
+                    ).agg(
                         {
                             "pipeline_version": lambda x: list(
                                 x.dropna().unique()
                             )
                         }
                     )
+                    # Turn indices from the groupby back into dataframe columns
                     .reset_index()
                 )
 
+                # Aggregate all completed pipelines for each session
+                session_grouped_data = pipeline_grouped_data.groupby(
+                    ["sub_id", "session_id", "session_type"],
+                )
                 session_completed_pipeline_data = (
-                    pipeline_grouped_data.groupby(
-                        ["sub_id", "session_id", "session_type"]
-                    )
-                    .apply(
-                        lambda x: dict(
-                            zip(x["pipeline_name"], x["pipeline_version"])
-                        )
+                    session_grouped_data.apply(
+                        lambda x: {
+                            pname: pvers
+                            for pname, pvers in zip(
+                                x["pipeline_name"], x["pipeline_version"]
+                            )
+                            if not pd.isnull(pname)
+                        }
                     )
+                    # NOTE: This expects a pd.Series and will not work on a pd.DataFrame
+                    # (pd.DataFrame.reset_index() doesn't have a "name" arg)
+                    # See related https://github.com/pandas-dev/pandas/issues/55225
```

Comment:
Can you explain how you could get a DataFrame here instead of a series? Maybe even linking
to the (soon to be closed) issue